pkg/operator/controller/nodeobservability/daemonset.go

```diff
@@ -31,19 +34,72 @@ const (
 	certsMountPath        = "/var/run/secrets/openshift.io/certs"
 )
 
+// currentNodeObsKubeletCAConfigMap gets the kubelet CA configmap resource.
+func (r *NodeObservabilityReconciler) currentNodeObsKubeletCAConfigMap(ctx context.Context, nsName types.NamespacedName) (bool, *corev1.ConfigMap, error) {
+	cm := &corev1.ConfigMap{}
+	if err := r.Get(ctx, nsName, cm); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, err
+	}
+
+	return true, cm, nil
+}
+
+// buildMapHash is a utility function to get a checksum of a data map.
+func buildMapHash(data map[string][]byte) (string, error) {
+	keys := make([]string, 0, len(data))
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	hash := sha256.New()
+	for _, k := range keys {
+		_, err := hash.Write([]byte(k))
+		if err != nil {
+			return "", err
+		}
+		_, err = hash.Write(data[k])
+		if err != nil {
+			return "", err
+		}
+	}
+	return hex.EncodeToString(hash.Sum(nil)), nil
+}
+
+func buildStringMapHash(data map[string]string) (string, error) {
+	m := map[string][]byte{}
+	for k, v := range data {
+		m[k] = []byte(v)
+	}
+	return buildMapHash(m)
+}
```

Comment:
I thought you wanted a dedicated PR for this. That's why you created a subtask, no?